pkg/storage/tree/treediff.go

```diff
@@ -0,0 +1,240 @@
+package tree
```

Comment:
Perhaps `tree_diff` is more appropriate file name.

Reply:
<img width="162" alt="Screen Shot 2021-07-25 at 11 08 48" src="https://user-images.githubusercontent.com/6618620/126887421-e983a2f1-6ced-4a7e-9668-3f432f804949.png">
<img width="152" alt="Screen Shot 2021-07-25 at 11 09 06" src="https://user-images.githubusercontent.com/6618620/126887426-99da5947-aecf-4039-85ea-327fce4f87ed.png">

Actually, there is a reason for calling it `treediff` instead of `tree_diff`:
- The code file and test files are placed together in the first way: tree.go -> tree_test.go -> treediff.go ...
- Instead of the second way: tree.go -> treediff.go -> tree_test.go ...

But I think it's a project preference, and I will follow the convention in our code.